### PR TITLE
Element selection in variable formatting

### PR DESCRIPTION
**Pull request by jl-wynen**
The next step for #1472.

This allows users to choose how many elements to show as well as which elements (first, last, mixed).

### CODE REVIEW

File: tests/format/format_test.py [hunk 1]
```diff
@@ -23,6 +23,85 @@ def test_variable_default(var):
...
+
+    var = sc.arange('x', 0)
+    assert '[]' in f'{var:{s}}'
+    assert '[]' in f'{var:{s}#6}'
+    assert '[]' in f'{var:{s}#0}'
+
+    var = sc.scalar(5)
+    assert '[5]' in f'{var:{s}}'
+    assert '[5]' in f'{var:{s}#2}'
+    assert '[...]' in f'{var:{s}#0}'
```
Comment (SimonHeybrock) [inline tests/format/format_test.py:50]: Can we avoid the `[]` for scalars? A 0-D numpy array also just prints a number.

Comment (jl-wynen) [reply tests/format/format_test.py:50]: We can. But then we should change this in the C++ formatter, too. Don't know how easy that is...

Comment (jl-wynen) [reply tests/format/format_test.py:50]: Done

File: tests/format/format_test.py [hunk 2]
```diff
@@ -23,6 +23,88 @@ def test_variable_default(var):
     assert '{:}'.format(var) == str(var)
 
 
+@pytest.mark.parametrize('s', ('^', ''))
+def test_variable_default_length_central(s):
+    var = sc.arange('x', 10)
```
Comment (SimonHeybrock) [inline tests/format/format_test.py:28]: What happens for 2-D or higher? Does it raise?

Comment (jl-wynen) [reply tests/format/format_test.py:28]: No, it flattens the array (same as `__str__`):
```python
In [3]: f'{sc.array(dims=["x", "y"], values=[[1, 2, 3], [4, 5, 6]]):<}'
Out[3]: '<scipp.Variable> (x: 2, y: 3)      int64  [dimensionless]  [1, 2, 3, 4, ...]'
```

**Review by SimonHeybrock — APPROVED** [review]
Should add something to the release notes? Is this findable in the docs (via the search), i.e., how will users find the correct docstring?

**Comment by jl-wynen** [comment]
> Should add something to the release notes? Is this findable in the docs (via the search), i.e., how will users find the correct docstring?

It's not yet findable. I was waiting to see if we want to implement the rest of https://github.com/scipp/scipp/issues/1472 and then document it all in one go.

File: docs/visualization/representations-and-tables.ipynb [hunk 3]
```diff
@@ -252,23 +252,154 @@
...
+   "outputs": [],
+   "source": [
+    "f'{var:#5}'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Configure how elements are formatted: (Note the double colon!)"
```
Comment (SimonHeybrock) [inline docs/visualization/representations-and-tables.ipynb:311]: Explain the meaning of the double colon briefly? I know details are below, but something mention like `:variable-spec:element-spec` here?

File: docs/visualization/representations-and-tables.ipynb [hunk 4]
```diff
@@ -252,23 +252,154 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "f'{var::.1e}'"
```
Comment (SimonHeybrock) [inline docs/visualization/representations-and-tables.ipynb:322]: Add a note that the bits after the second `:` is just the normal spec that you can use in Python (and add a link?)

**Comment by jl-wynen** [comment]
@SimonHeybrock Do you have any more comments?

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**